vulnerabilities/models.py

```diff
@@ -200,6 +204,34 @@ class Meta:
     def __str__(self):
         return self.vulnerability_id
 
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+
+    @property
+    def get_date_published(self):
+        """
+        Return datetime object ``date_published`` for a vulnerability
+        If a vulnerability is created by NVD Importer then use NVD Importer's
+        date of publication as the publishing date of vulnerability.
+        Otherwise get the oldest date of publication.
+        """
+        from vulnerabilities.importers.nvd import NVDImporter
+        nvd_date = None
+        date_published = None
+        for advisory in self.advisories.all():
```

Comment:
Use filter, for created_by = NVD and date_published != None, sort by latest date_collected, to get the latest updates.